18843 Mobile and Pervasive Computing (#73)
CMU-HKN/CMU-ECE-CS-Guide: electives/18843.md

# 15-821/18-843: Mobile and Pervasive Computing

| Category     | Difficulty |
|:--           | :-:        |

> | Quizes       | 8          |

Reviewer: ```suggestion
| Quizzes       | 8          |
```

| Presentation | 7          |
| Project      | 8          |

> This course explores the principles and practice of mobile computing and its close relative, pervasive computing (aka “Internet of Things (IoT)”). The course will offer significant hands-on experience: in small teams, students will work under the guidance of a mentor on a project. Students will also, in small teams, present a summary and overview of the commercial landscape for one of the topics covered in class. There will be a brief quiz at the start of each class, based on the readings for that classes the principles and practice of mobile computing and its close relative, pervasive computing (aka “Internet of Things (IoT)”).

Reviewer: The latter part of the last sentence doesn't make sense

Reviewer: `based on the readings for that classes the principles and practice of mobile computing and its close relative, pervasive computing (aka “Internet of Things (IoT)”).`

Author: I will change the sentence, although that one specifically was taken from his website.

Author: Updated last sentence.

## What to expect

>  - There are a total of 12 units. There will be an average of 6 papers to read for the following class. At the beginning of each class there will be a quiz based on the previous weeks readings

Reviewer: ```suggestion
 - There are a total of 12 units. There will be an average of 6 papers to read for the following class. At the beginning of each class there will be a quiz based on the previous week's readings
```

Author: Regarding the difficulty, that is a very subjective thing, I honestly don't understand why you include such a category. Anyways, the quizzes really add up with the rest of the workload (from other classes) and they are closed book and the questions tend to be very specific so you almost need to study the papers. The commercial scan is fairly straightforward but it takes some work in advance to be able to ace it. And the project is quite binary, either too difficult or fairly easy. My project specifically was about Neural Radiance Fields and it had nothing to do with the class curriculum and I had never done it before. Plus, my mentor didn't help at all. So, if you choose an easy project and the mentor is supportive then it's a difficulty of 6 but if you happen to choose a difficult project and the mentor doesn't get themselves engaged in the project then is an 8.

Reviewer: Sure, difficulty is subjective but I thought it would be useful to get an idea of "generally" how difficult a class is. E.g. some classes are generally easy and some are generally hard. It's not a perfect metric, but I do feel like it's better than nothing. E.g. some people step into classes thinking it will be ok but end up struggling because of unexpected workloads.

Author: Updated it.

 - You are expected to go to class and the quizzes can only be done in class and not remotely
 - There is no textbook, all papers can be found online
 - **There is no mid-term nor final exam**
 - There is a semester long project in pairs, which is chosen the first day of classes 
 - There is also a "Commercial Scan" presentation in which you research and present a relevant state of the art in industry, and how commercial practice diverges from academic research

## Grading weights
 - Project (55%)
 - Commercial scan presentation (20%)

>  - Quizes (20%) -- you drop the worst test score

Reviewer: ```suggestion
 - Quizzes (20%) -- you drop the worst test score
```

 - Class participation (5%)

## What to watch out for

>  - The quizes are clossed book so it's important to read them well to get the general idea but also take notes since the quizes tend to ask fine grained details

Reviewer: ```suggestion
 - The quizzes are closed-book so it's important to read them well to get the general idea but also take notes since the quizzes tend to ask fine grained details
```
